Core/Filesystem/nandfs.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (C) 2022  University of Alberta
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+/*
+ * nandfs.c
+ *
+ *  Created on: Jun 14, 2022
+ *      Author: Robert Taylor
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+#include "nandfs.h"
+#include "nand_errno.h"
+#include "nand_m79a_lld.h"
+#include "nand_core.h"
+
+int nand_errno = 0;
+
+FileHandle_t handles[FILEHANDLE_COUNT];
+DirHandle_t dir_handles[DIRHANDLE_COUNT];
+
+/*
+ * Private function. Returns pointer to unopened file handle
+ */
+static FileHandle_t *_get_handle() {
+	for (int i = 0; i < FILEHANDLE_COUNT; i++) {
+		if (handles[i].open) {
+			continue;
+		} else {
+			return &(handles[i]);
+		}
+	}
+	return -1;
+}
+
+/*
+ * Private function. Returns pointer to unopened directory handle
+ */
+static DirHandle_t *_get_dir_handle() {
+	for (int i = 0; i < DIRHANDLE_COUNT; i++) {
+		if (dir_handles[i].open) {
+			continue;
+		} else {
+			return &(dir_handles[i]);
+		}
+	}
+	return -1;
+}
+
+int NANDfs_init() {
```

Comment:
Yo dawg, i heard you like wrapper functions, so I wrapped your wrapper function so your wrapper function has a wrapper function